joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/TypeInferenceTests.scala

```diff
@@ -197,6 +199,58 @@ class NewTypeInferenceTests extends JavaSrcCode2CpgFixture {
       }
     }
   }
+
+  "chained calls from external dependencies" should {
+    lazy val cpg = code(
+      """
+        |package net.javaguides.hibernate;
+        |
+        |import java.util.List;
+        |
+        |import org.hibernate.Session;
+        |import org.hibernate.Transaction;
+        |
+        |import net.javaguides.hibernate.entity.Student;
+        |import net.javaguides.hibernate.util.HibernateUtil;
+        |
+        |public class NamedQueryExample {
+        |	public static void main(String[] args) {
+        |		saveStudent();
+        |
+        |		Transaction transaction = null;
+        |		try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+        |			transaction = session.beginTransaction();
+        |			// Executing named queries
+        |
+        |			List<Long> totalStudents = session.createNamedQuery("GET_STUDENTS_COUNT", Long.class).getResultList();
+        |			System.out.println("Total Students: " + totalStudents.get(0));
+        |
+        |			transaction.commit();
+        |		} catch (Exception e) {
+        |			if (transaction != null) {
+        |				transaction.rollback();
+        |			}
+        |		}
+        |
+        |	}
+        |}
+        |""".stripMargin,
+      Seq("net", "javaguides", "hibernate", "NamedQueryExample.java").mkString(File.separator)
+    )
+
+    "should be resolved using dummy return values" in {
+      val Some(getResultList) = cpg.call("getResultList").headOption
+      // Changes the below from <unresolvedNamespace>.getResultList:<unresolvedSignature>(0) to:
+      getResultList.methodFullName shouldBe "org.hibernate.Session.createNamedQuery:<unresolvedSignature>(2).<returnValue>.getResultList"
```

Comment:
For Java, it could be useful to preserve the signature of `getResultList`. I'm not sure this would actually be useful at this level of uncertainty, but it is information that could potentially disambiguate fullNames.

Reply:
@johannescoetzee I see, so this would become more like `org.hibernate.Session.createNamedQuery:<unresolvedSignature>(2).<returnValue>.getResultList:<unresolvedSignature>(0)` in this case?

Reply:
Indeed, although as I mentioned above, I'm not sure this would be useful information. By the time we get to this point, I don't imagine we'd be able to say much more than "this call goes to library X" in any case, so treating `getResultList:<unresolvedSignature>(0)` and `getResultList:<unresolvedSignature>(1)` as distinct is unlikely to tell us anything useful.